pkg/terraform/terraform.go

```diff
@@ -23,27 +24,49 @@ const (
 
 	// VarFileName is the default name for Terraform var file.
 	VarFileName string = "terraform.tfvars"
+
+	// TargetBootstrap is the default name for the bootstrap directory
+	TargetBootstrap string = "bootstrap"
+
+	// TargetCompat is the name for the legacy terraform directory
+	TargetCompat string = "compat"
 )
 
+// GetStateFileName returns the name of the terraform state file. If no target
+// is specified or compat is specified, default to the old state file name.
+func GetStateFileName(target string) string {
+	if target != "" && target != TargetCompat && !LegacyMode() {
```

Comment:
Why do we need to check for legacy mode?

Reply:
Legacy mode was supposed to determine if running in legacy mode ;-) It's not so important anymore so I've removed it.